app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestPresenter.java

```diff
@@ -65,4 +72,22 @@ public void onNewCategoryAdded(String categoryName, String categoryDescription)
     public void showAddCategoryScreen() {
         view.navigateToAddCategoryPage(contest);
     }
+
+    void submitContest() {
+        view.showMessage(R.string.submitting_contest);
+        Observable<ContestWrapper> call = RetrofitClient.getApi()
+                .submitContest(new ContestWrapper(contest.build()));
+        call.compose(subscribeOnIoObserveOnUi())
```

Comment:
instead of doing this, you can chain the `compose` operator to `submitContest()` and change the type of `call` to `Disposable`